include/circt/Dialect/FIRRTL/FIRRTLOpInterfaces.td

```diff
@@ -262,20 +267,20 @@ def FModuleLike : OpInterface<"FModuleLike"> {
       return "portSyms";
     }
 
-    /// Replace NUll entries with invalid sym and clear array if all elements
+    /// Replace NULL entries with invalid sym and clear array if all elements
     /// are invalid.
     static void fixupPortSymsArray(SmallVectorImpl<Attribute> &syms,
                                    MLIRContext *context) {
-      bool isValid = false;
-      for (size_t i = 0, e = syms.size(); i < e; ++i)
-        if (!syms[i])
-          syms[i] = InnerSymAttr::get(context);
-        else
-          isValid = true;
       // The lack of *any* port symbol is represented by an empty `portSyms`
       // array as a shorthand.
-      if (!isValid)
+      if (llvm::none_of(syms, [&](auto &sym) { return sym; })) {
```

Comment:
Should this also be checking for an empty `InnerSymAttr`?

Reply:
I'm not sure, I forget where things landed there.  I think so offhand, but should check.

Reply:
I'm not sure anymore, to be honest.  Looking elsewhere we do check both if it's "null" (`operator bool` on Attribute) as well as whether it's `empty()` after casting to `InnerSymAttr`.  I guess this should as well.  Is that right, @prithayan , or especially does that seem wrong? :smile: .

Reply:
Added the check: https://github.com/llvm/circt/pull/4223/commits/be714b6e7d866b6b2fef5e21218390f90f4fd8b0 .